nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -17,6 +17,9 @@ let
   cfgZED = config.services.zfs.zed;
 
   selectModulePackage = package: config.boot.kernelPackages.${package.kernelModuleAttribute};
+  clevisDatasets = map (e: e.device) (filter (e: (hasAttr e.device config.boot.initrd.clevis.devices) && e.fsType == "zfs" && (fsNeededForBoot e)) config.system.build.fileSystems);
```

Comment:
I think this is mishandling fileSystems with `device = null` (the default).

For example, I'm mounting `/boot` by label with

```
  fileSystems."/boot" =
    { label = "boot";
      fsType = "ext4";
    };
```

Reply:
Yes, I think you are right